packages/openapi-generator/src/parser/operation-naming.ts

```diff
@@ -0,0 +1,178 @@
+import {
+  partition,
+  UniqueNameGenerator,
+  pascalCase,
+  camelCase
+} from '@sap-cloud-sdk/util';
+import { Method } from '../openapi-types';
+import { operationNameExtension } from '../extensions';
+import { OperationInfo } from './operation-info';
+
+/**
+ * Modify each operation to contain a unique `operationId`.
+ * @param openApiDocument OpenAPI JSON document.
+ * @returns The modified document.
+ */
+export function ensureUniqueOperationIds(
+  operations: OperationInfo[]
+): OperationInfo[] {
+  const {
+    uniqueOperationNames,
+    operationsWithUniqueNames,
+    operationsWithDuplicateNames
+  } = partitionNamedOperationsToUniqueAndDuplicate(operations);
+
+  const nameGenerator = new UniqueNameGenerator('', uniqueOperationNames);
+
+  const renamedOperations = renameOperations(
+    operationsWithDuplicateNames,
+    ({ operation }) =>
+      nameGenerator.generateAndSaveUniqueName(camelCase(operation.operationId!))
+  );
+  return [...operationsWithUniqueNames, ...renamedOperations];
+}
+
+export function nameOperations(operations: OperationInfo[]): OperationInfo[] {
+  const [operationsWithExtensions, operationsWithoutExtensions] = partition(
+    operations,
+    ({ operation }) => !!operation[operationNameExtension]
+  );
+
+  validateUniqueExtensions(operationsWithExtensions);
+
+  const renamedOperationsWithExtensions = renameOperations(
+    operationsWithExtensions,
+    renameOperationWithExtension
+  );
+
+  const [namedOperations, unnamedOperations] = partition(
+    operationsWithoutExtensions,
+    ({ operation }) => !!operation.operationId
+  );
+
+  const renamedUnnamedOperations = renameOperations(
+    unnamedOperations,
+    ({ pathPattern, method }) =>
+      getOperationNameFromPatternAndMethod(pathPattern, method)
+  );
+
+  return [
+    ...renamedOperationsWithExtensions,
+    ...namedOperations,
+    ...renamedUnnamedOperations
+  ];
+}
+
+function validateUniqueExtensions(operations: OperationInfo[]): void {
+  operations.reduce((uniqueOperationNames: string[], operationInfo) => {
+    const operationName = renameOperationWithExtension(operationInfo);
```

Comment:
I would not expect the the operations are modified (renamed) in a `validate` function. Maybe use something like "rename"/"format".

Reply:
This is separate. We only get the name and add it to the list of unique names.